components/TabList/tab-list.tsx

```diff
@@ -77,6 +77,11 @@ export const SubTabsList = ({ tabList, pageId, selectedTab, label, textSize, onS
             key={index}
             className={clsx(isSelected && "bg-white shadow", "rounded py-1 px-2", !isSelected && "text-light-slate-11")}
             onClick={onSelect ? () => onSelect(tab) : undefined}
+            onKeyDown={(e) => {
```

Comment:
For the SubTabsList, it's not a tab list if there is no `onSelect`, it's just links. But if the `onSelect` is present, it means it is a tab list. We can rework the SubTabsList component to this.

```typescript
export const SubTabsList = ({ tabList, pageId, selectedTab, label, textSize, onSelect }: SubTabsListProps) => {
  const ContainerComponent = onSelect ? "div" : "nav";
  const extraProps = onSelect ? { role: "tablist" } : {};

  return (
    <ContainerComponent
      {...extraProps}
      aria-label={label}
      className={clsx(
        "flex items-center w-max overflow-x-auto overflow-y-hidden gap-2 bg-light-slate-3 p-1 rounded",
        textSize === "small" && "text-sm"
      )}
    >
      {tabList.map((tab, index) => {
        const isSelected = selectedTab === tab.name.toLowerCase();

        return (
          <div
            key={index}
            className={clsx(isSelected && "bg-white shadow", "rounded py-1 px-2", !isSelected && "text-light-slate-11")}
          >
            {onSelect ? (
              <button role="tab" onClick={() => onSelect(tab)}>
                tab.name
              </button>
            ) : (
              <Link href={`${pageId ? `${pageId}/` : ""}${tab.path}`}>{tab.name}</Link>
            )}
          </div>
        );
      })}
    </ContainerComponent>
  );
};
```

This could be improved, but I don't want to introduced too much refactoring for the moment as the goal of the issue was to sort out some accessibility issues.

Note that because we've introduced a button when the `onSelect` exists, i.e. 

```
            {onSelect ? (
              <button role="tab" onClick={() => onSelect(tab)}>
                tab.name
              </button>
            ) : (
              <Link href={`${pageId ? `${pageId}/` : ""}${tab.path}`}>{tab.name}</Link>
            )}
```

we no longer need the `onKeyDown` to handle <kbd>ENTER</kbd> as this is handled natively with a button.